plugins/org.jkiss.dbeaver.erd.ui/src/org/jkiss/dbeaver/erd/ui/editor/ERDEditorStandalone.java

```diff
@@ -125,17 +126,25 @@ public void doSave(IProgressMonitor monitor)
     {
         try {
             String diagramState = DiagramLoader.serializeDiagram(RuntimeUtils.makeMonitor(monitor), getDiagramPart(), getDiagram(), false, false);
-
-            final IFile file = EditorUtils.getFileFromInput(getEditorInput());
-            if (file == null) {
-                throw new DBException("Can't determine diagram file");
+            IEditorInput editorInput = getEditorInput();
+            final IFile file = EditorUtils.getFileFromInput(editorInput);
+            if (file != null) {
+                file.setContents(
+                    new ByteArrayInputStream(diagramState.getBytes(StandardCharsets.UTF_8)),
+                    true,
+                    true,
+                    monitor);
+                getCommandStack().markSaveLocation();
```

Comment:
This can be moved after the `if` chain to avoid duplicating code

Reply:
Generally, this block can not be after if because in the first case a file is IFile under EFS, in the second case File - java.io 
both have different API for write content.

Reply:
I'm talking solely about the `getCommandStack().markSaveLocation();` statement. It's present in both cases, and the third one throws an exception (hence unreachable), so it can be safely moved below

Reply:
ok, get it.